### PR TITLE
Update aspnetcore instrumentation benchmarks

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #.

## Changes
Adding updated benchmarks for AspNetCore Instrumentation using minimal hosting model.  If these look good, I will remove the existing benchmarks in a follow up.
**Results:**
BenchmarkDotNet=v0.13.1, OS=Windows 10.0.22000
Intel Core i7-8850H CPU 2.60GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
.NET SDK=7.0.100-preview.6.22275.1
  [Host] : .NET 6.0.8 (6.0.822.36306), X64 RyuJIT
Job=InProcess  Toolchain=InProcessEmitToolchain
|                                      Method |     Mean |   Error |  StdDev |  Gen 0 | Allocated |
|-------------------------------------------- |---------:|--------:|--------:|-------:|----------:|
|                 UninstrumentedAspNetCoreApp | 155.6 us | 2.63 us | 2.33 us | 0.9766 |      5 KB |
| InstrumentedAspNetCoreAppWithDefaultOptions | 176.8 us | 3.24 us | 2.70 us | 1.2207 |      7 KB |

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        private HttpClient httpClient;
+        private WebApplication app;
+        private TracerProvider tracerProvider;
+
+        [GlobalSetup(Target = nameof(UninstrumentedAspNetCoreApp))]
+        public void UninstrumentedAspNetCoreAppGlobalSetup()
+        {
+            var builder = WebApplication.CreateBuilder();
+            builder.Services.AddControllers();
+            builder.Logging.ClearProviders();
```
Comment (cijothomas) [inline test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs:57]: do we need to?

File: test/Benchmarks/Helper/ValuesController.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        // GET api/values/5
+        [HttpGet("{id}")]
+        public string Get(int id)
+        {
+            return "value";
+        }
+
+        // POST api/values
```
Comment (cijothomas) [inline test/Benchmarks/Helper/ValuesController.cs:40]: remove everything not used by benchmarks.

**Comment by cijothomas** [comment]
Please run the benchmarks and share the numbers. Can you confirm if the tracerprovider is only active for the non-instrumented scenario?

File: test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs (hunk 1 above)
Comment (vishweshbankwar) [reply test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs:57]: Yes - lot of internal logging happens while running the benchmarks (for each request).

**Comment by vishweshbankwar** [comment]
> Please run the benchmarks and share the numbers. Can you confirm if the tracerprovider is only active for the non-instrumented scenario?

@cijothomas - The numbers are added on the top of class file. Update it in the description as well.

File: test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs (hunk 1 above)
Comment (cijothomas) [reply test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs:57]: if we are doing the same for both instrumented and uninstrumented, we are good. Maybe its a good idea to extract this common logic to a helper method, so we know it'll be consistent.

**Review by utpilla — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**